x-pack/plugins/observability_solution/observability/public/pages/overview/components/sections/metrics/host_link.tsx

```diff
@@ -5,19 +5,38 @@
  * 2.0.
  */
 import React from 'react';
+import {
+  ASSET_DETAILS_LOCATOR_ID,
+  AssetDetailsLocatorParams,
+} from '@kbn/observability-shared-plugin/common';
+import { useLocatorRedirect } from '../../../../../hooks/use_locator';
 import { StringOrNull } from '../../../../..';
 
 interface Props {
   name: StringOrNull;
-  id: StringOrNull;
-  timerange: { from: number; to: number };
+  id: string;
+  timerange: { from?: number; to?: number };
 }
 
 export function HostLink({ name, id, timerange }: Props) {
-  const link = `../../app/metrics/link-to/host-detail/${id}?from=${timerange.from}&to=${timerange.to}`;
+  const { getRedirectUrl } =
+    useLocatorRedirect<AssetDetailsLocatorParams>(ASSET_DETAILS_LOCATOR_ID);
```

Comment:
Q: why don't we use the locator directly instead of this abstraction?

Reply:
We could, I was assuming here we might want to reuse this in other places where we might need to use locators/replace `link-to`. If not necessary, I could just merge this into `HostLink`